target_postgres/sinks.py

```diff
@@ -324,6 +324,15 @@ def activate_version(self, new_version: int) -> None:
             )
             return
 
+        if self._pending_batch:
+            self.logger.info(
+                "An activate version message for '%s' was received. Draining...",
+                self.stream_name,
+            )
+            draining_status = self.start_drain()
+            self.process_batch(draining_status)
+            self.mark_drained()
```

Comment:
I'm taking a note to make sure we do this upstream: https://github.com/meltano/sdk/issues/2576